### PR TITLE
[Feature][RestAPI] Support submit job with seatunnel style hocon format config

### LINKED ISSUE #7925: [Feature][RestAPI] Support submit job with seatunnel style hocon format config
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

At now, user only can use json format to submit job with rest api. But json format not fisrt choice in SeaTunnel. We should support submit job with seatunnel style hocon format in rest api.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by CosmosNi**
#7925

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by Hisoka-X** [comment]
cc @liugddx

**Review by Hisoka-X — COMMENTED** [review]
Please update the docs. Thanks @CosmosNi !

**Comment by liugddx** [comment]
Thanks @CosmosNi ! Can you add a test case for normal submission?  like https://github.com/apache/seatunnel/blob/ccd6a2a11e8d8198258301c81d71e8ed306e917a/seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterSeaTunnelContainer.java#L123C17-L123C45

**Comment by liunaijie** [comment]
I see this pr is put the config string to request body.
Do we conside using `multipart/form-data` upload the file into a `config_file` (or other name) parameter

**Comment by Hisoka-X** [comment]
> Do we conside using multipart/form-data upload the file into a config_file (or other name) parameter

Ok for me, but I think this should be another feature. Please create an issue for this.

**Comment by liunaijie** [comment]
> > Do we conside using multipart/form-data upload the file into a config_file (or other name) parameter
> 
> Ok for me, but I think this should be another feature. Please create an issue for this.

Yes, it should be in another pr, just point it in this issue.

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestConstant.java [hunk 1]
```diff
@@ -27,6 +27,8 @@ public class RestConstant {
 
     public static final String IS_STOP_WITH_SAVE_POINT = "isStopWithSavePoint";
 
+    public static final String ENV_CONFIG_STYLE = "configStyle";
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestConstant.java]:
```suggestion
    public static final String ENV_CONFIG_FORMAT = "format";
```
I think use the format more clear.

**Review by Hisoka-X — COMMENTED** [review]
Thanks @CosmosNi ! Please update the doc too.

File: docs/en/seatunnel-engine/rest-api-v2.md [hunk 2]
```diff
@@ -384,11 +384,12 @@ When we can't get the job info, the response will be:
...
 #### Parameters
 
-> |         name         |   type   | data type |            description            |
-> |----------------------|----------|-----------|-----------------------------------|
+> | name                 |   type   | data type |            description            |
+----------------------|----------------------|----------|-----------|-----------------------------------|
 > | jobId                | optional | string    | job id                            |
 > | jobName              | optional | string    | job name                          |
 > | isStartWithSavePoint | optional | string    | if job is started with save point |
+> | format               | optional | string    | config style,json/hocon,default json |
```
Comment (Hisoka-X) [inline docs/en/seatunnel-engine/rest-api-v2.md]:
```suggestion
> | format               | optional | string    | config format, support json and hocon, default json |
```

**Comment by CosmosNi** [comment]
@Hisoka-X @liugddx PTAL，thanks

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestConstant.java [hunk 3]
```diff
@@ -27,6 +27,8 @@ public class RestConstant {
 
     public static final String IS_STOP_WITH_SAVE_POINT = "isStopWithSavePoint";
 
+    public static final String ENV_CONFIG_STYLE = "format";
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestConstant.java]:
```suggestion
    public static final String CONFIG_FORMAT = "format";
```

File: docs/en/seatunnel-engine/rest-api-v2.md [hunk 4]
```diff
@@ -384,15 +384,17 @@ When we can't get the job info, the response will be:
 
 #### Parameters
 
-> |         name         |   type   | data type |            description            |
-> |----------------------|----------|-----------|-----------------------------------|
+> | name                 |   type   | data type |            description            |
+----------------------|----------------------|----------|-----------|-----------------------------------|
```
Comment (Hisoka-X) [inline docs/en/seatunnel-engine/rest-api-v2.md]:
```suggestion
> |----------------------|----------|-----------|-----------------------------------|
```
The markdown format not right.
image

File: docs/en/seatunnel-engine/rest-api-v2.md [hunk 5]
```diff
@@ -384,15 +384,17 @@ When we can't get the job info, the response will be:
...
 > | jobId                | optional | string    | job id                            |
 > | jobName              | optional | string    | job name                          |
 > | isStartWithSavePoint | optional | string    | if job is started with save point |
+> | format               | optional | string    | config format, support json and hocon, default json |
 
 #### Body
 
-```json
+You can choose json or hocon to pass request body.Examples are as follows:
+``` json example
```
Comment (Hisoka-X) [inline docs/en/seatunnel-engine/rest-api-v2.md]:
```suggestion
The json format example:
``` json
```

File: docs/en/seatunnel-engine/rest-api-v2.md [hunk 6]
```diff
@@ -384,15 +384,17 @@ When we can't get the job info, the response will be:
...
+----------------------|----------------------|----------|-----------|-----------------------------------|
 > | jobId                | optional | string    | job id                            |
 > | jobName              | optional | string    | job name                          |
 > | isStartWithSavePoint | optional | string    | if job is started with save point |
+> | format               | optional | string    | config format, support json and hocon, default json |
 
 #### Body
 
-```json
+You can choose json or hocon to pass request body.Examples are as follows:
```
Comment (Hisoka-X) [inline docs/en/seatunnel-engine/rest-api-v2.md]:
```suggestion
You can choose json or hocon to pass request body.
```

File: docs/en/seatunnel-engine/rest-api-v2.md [hunk 7]
```diff
@@ -421,6 +423,36 @@ When we can't get the job info, the response will be:
     ]
 }
 ```
+``` hocon example
```
Comment (Hisoka-X) [inline docs/en/seatunnel-engine/rest-api-v2.md]:
```suggestion
The hocon format example:
``` hocon
```

File: docs/zh/seatunnel-engine/rest-api-v2.md [hunk 8]
```diff
@@ -380,15 +380,17 @@ seatunnel:
...
 > | jobId                | optional | string | job id                            |
 > | jobName              | optional | string | job name                          |
 > | isStartWithSavePoint | optional | string | if job is started with save point |
+> | format               | optional | string    | 配置风格,支持json和hocon,默认 json         |
 
 #### 请求体
 
-```json
+你可以选择用json或者hocon的方式来传递请求体。示例如下：
+```json 示例
```
Comment (Hisoka-X) [inline docs/zh/seatunnel-engine/rest-api-v2.md]:
```suggestion
你可以选择用json或者hocon的方式来传递请求体。
Json请求示例：
```json
```

File: docs/zh/seatunnel-engine/rest-api-v2.md [hunk 9]
```diff
@@ -418,6 +420,35 @@ seatunnel:
 }
 ```
 
+```hocon示例
```
Comment (Hisoka-X) [inline docs/zh/seatunnel-engine/rest-api-v2.md]:
```suggestion
Hocon请求示例：
```hocon
```

**Review by Hisoka-X — DISMISSED** [review]
LGTM if ci passes. Thanks @CosmosNi ! cc @liugddx

**Review by Hisoka-X — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Comment by liugddx** [comment]
LGTM Thanks @CosmosNi !

**Pull request merged**